astilectron.go

```diff
@@ -194,35 +199,90 @@ func (a *Astilectron) provision() error {
 	return a.provisioner.Provision(ctx, a.options.AppName, runtime.GOOS, runtime.GOARCH, *a.paths)
 }
 
-// listenTCP creates a TCP server for astilectron to connect to
-// and listens to the first TCP connection coming its way (this should be Astilectron).
-func (a *Astilectron) listenTCP() (err error) {
+// function to create TCP/Unix socket connection
+func (a *Astilectron) listen() (string, error) {
```

Comment:
After seeing this, it seems overly complicated.

Here's what I'd rather do:

1) Determine `var addr string` => check `a.options.Addr` first, then check `os`
2)  Depending on the `scheme` of `addr`, create the proper connection